src/metabase/search/ingestion.clj

```diff
@@ -115,25 +115,43 @@
   false)
 
 (defn update!
-  "Update all active engines' existing indexes with the given documents"
-  [documents-reducible]
+  "Update all active engines' existing indexes with the given documents. Passed remove-documents will be deleted from the index."
+  [documents-reducible remove-documents]
   (when-let [engines (seq (search.engine/active-engines))]
     (if (= 1 (count engines))
-      (search.engine/update! (first engines) documents-reducible)
-      ;; TODO um, multiplexing over the reducible awkwardly feels strange. We at least use a magic number for now.
-      (doseq [batch (eduction (partition-all 150) documents-reducible)
-              e     engines]
-        (search.engine/update! e batch)))))
+      (u/prog1 (search.engine/update! (first engines) documents-reducible)
+        (doseq [[group ids] (u/group-by first second remove-documents)]
+          (search.engine/delete! (first engines) group ids)))
+      (u/prog1
+        ;; TODO um, multiplexing over the reducible awkwardly feels strange. We at least use a magic number for now.
```

Comment:
I forgot why we needed to do this when I first looked at the code.

It's because of two potential issues with consuming the reducible:

1. We may only be able to iterate the reducible ones (like java streams)
2. Holding the reference to traverse the second time could blow things up (head holding)

At least I think that's why, forced to speculate cos this comment sucks, let's flesh it out a bit more.